client/product_details/Testing/endToEndTests.test.js

```diff
@@ -39,85 +39,85 @@ afterEach(async () => {
     await page.goto(testUrl);
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-    await page.click('#upScroll-container');
-
-    const fifth = await page.$$('#5');
-    const sixth = await page.$$('#6');
-    const seventh = await page.$$('#7');
-    const eighth = await page.$$('#8');
-    const nineth = await page.$$('#9');
-    const tenth = await page.$$('#10');
-    const initial = await page.$$('#0');
-
+    var upScroll = await page.waitForSelector('#upScroll-container');
+    await upScroll.click();
+    await upScroll.click();
+    await upScroll.click();
+    await upScroll.click();
+    await upScroll.click();
+    await upScroll.click();
+    await upScroll.click();
+    const fifth = page.$eval("[id='5']", el => el ? true : false);
+    const sixth = page.$eval("[id='6']", el => el ? true : false);
+    const seventh = page.$eval("[id='7']", el => el ? true : false);
+    const eighth = page.$eval("[id='8']", el => el ? true : false);
+    const nineth = page.$eval("[id='9']", el => el ? true : false);
+    const tenth = page.$eval("[id='10']", el => el ? true : false);
+    const initial = page.$eval("[id='0']", el => el ? true : false);
     expect(fifth).toBeTruthy();
     expect(sixth).toBeTruthy();
     expect(seventh).toBeTruthy();
     expect(eighth).toBeTruthy();
     expect(nineth).toBeTruthy();
     expect(tenth).toBeTruthy();
     expect(initial).toBeTruthy();
+
   })
   it('should scroll correctly after five down-clicks', async () => {
     await page.goto(testUrl);
-    await page.click('#downScroll-container');
-    await page.click('#downScroll-container');
-    await page.click('#downScroll-container');
-    await page.click('#downScroll-container');
-    await page.click('#downScroll-container');
-    
-    const fifth = await page.$$('#5');
-    const sixth = await page.$$('#6');
-    const seventh = await page.$$('#7');
-    const eighth = await page.$$('#8');
-    const nineth = await page.$$('#9');
-    const tenth = await page.$$('#10');
-    const initial = await page.$$('#0');
-    
+    const downScroll = await page.waitForSelector('#downScroll-container');
+    await downScroll.click();
+    await downScroll.click();
+    await downScroll.click();
+    await downScroll.click();
+    await downScroll.click();
+    const fifth = page.$eval("[id='5']", el => el ? true : false);
+    const sixth = page.$eval("[id='6']", el => el ? true : false);
+    const seventh = page.$eval("[id='7']", el => el ? true : false);
+    const eighth = page.$eval("[id='8']", el => el ? true : false);
+    const nineth = page.$eval("[id='9']", el => el ? true : false);
+    const tenth = page.$eval("[id='10']", el => el ? true : false);
+    const initial = page.$eval("[id='0']", el => el ? true : false);
```

Comment:
I don't have anything much to add for your Puppeteer testing except that I would totally love to borrow some of these at some point if that's alright with you, super neat!

Reply:
Aww! Feel free to borrow any and all of it :)